pkg/registry/registry_test.go

```diff
@@ -793,8 +795,13 @@ func TestRegistryClient_DownloadStarterProject(t *testing.T) {
 				preferenceClient: tt.fields.preferenceClient,
 				kubeClient:       tt.fields.kubeClient,
 			}
-			if err := o.DownloadStarterProject(tt.args.starterProject, tt.args.decryptedToken, tt.args.contextDir, tt.args.verbose); (err != nil) != tt.wantErr {
-				t.Errorf("DownloadStarterProject() error = %v, wantErr %v", err, tt.wantErr)
+			gotContainsDevfile, gotErr := o.DownloadStarterProject(tt.args.starterProject, tt.args.decryptedToken, tt.args.contextDir, tt.args.verbose)
+			if (gotErr != nil) != tt.wantErr {
+				t.Errorf("DownloadStarterProject() error = %v, wantErr %v", gotErr, tt.wantErr)
+				return
+			}
+			if tt.wantContainsDevfile && !gotContainsDevfile {
```

Comment:
Why don't you test an exact match? Otherwise, we might not check some cases (e.g., if `tt.wantContainsDevfile` is `false`, but `gotContainsDevfile` is `true`).

```suggestion
			if tt.wantContainsDevfile != gotContainsDevfile {
```